app/controllers/users_controller.rb

```diff
@@ -0,0 +1,62 @@
+class UsersController < ApplicationController
+  before_action :authorize_user, only: [:index, :edit, :update, :destroy]
+  before_action :check_current_user,   only: [:show, :edit, :update]
+  before_action :authorize_admin,     only: :destroy
+
+  def show
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      log_in @user
+      flash[:success] = "Welcome to the Eureka Caffee!"
+      redirect_to @user
+    else
+      flash.now[:danger] = 'Please verify the information and try again.'
+      render 'new'
+    end
+  end
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = "Profile updated"
+      redirect_to @user
+    else
+      render 'edit'
+    end
+  end
+
+  private
+
```

Comment:
small trick:

create a 

```
def set_user
  @user = User.find(params[:id])
end
```

add `before_filter set_user, only: [:edit, :update]` wherever you need the user and remove that line, so we keep it DRY (Do Not repeat Yourself)

Reply:
solved. ✓